src/components/Row/LinkSegmentBuilder.jsx

```diff
@@ -0,0 +1,82 @@
+import cx from "classnames";
+import { HyperLinkPreview } from "../../js/image_preview";
+import HyperLink from "./HyperLink";
+import ColorSegmentBuilder from "./ColorSegmentBuilder";
+
+export class LinkSegmentBuilder {
+  constructor(row, showsLinkPreviews, forceWidth, highlighted) {
+    this.row = row;
+    this.forceWidth = forceWidth;
+    this.highlighted;
+    //
+    this.segs = [];
+    this.linkPreviews = showsLinkPreviews ? [] : false;
+    //
+    this.colorSegBuilder = null;
+    this.col = null;
+    this.href = null;
+  }
+
+  beginSegment() {
```

Comment:
This pushes a segment into the array. Maybe name it `saveSegment` or something?

Reply:
Good suggestion. Renamed